gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1338,6 +1340,17 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 								n.Args[1] = args1
 							}
 						}
+					} else if fv.PkgPath == uversePkgPath && fv.Name == "new" {
+						// new returns a pointer and pointers are always addressable.
+						n.IsAddressable = true
+					}
+				}
+
+				if !n.IsAddressable && len(ft.Results) == 1 {
+					baseType := baseOf(ft.Results[0].Type)
+					switch baseType.(type) {
+					case *PointerType, *SliceType:
+						n.IsAddressable = true
```

Comment:
I know this is out of scope, but the error should be about assignment mismatch.

```go
package main

func foo() ([]int, []string) {

	return []int{1, 2, 3}, []string{"a", "b", "c"}
}

func main() {

	r1 := &foo()

}

// Error:
// cannot take address of foo<VPBlock(3,0)>():
```

Reply:
This one is actually in scope; you're right that in this case it doesn't make sense for the addressability error message to supersede the results mismatch. In fact, the addressability message doesn't make sense because it is trying to take the address of multiple values.  Addressed this here: https://github.com/gnolang/gno/pull/2731/commits/3f4f1b6ee9c516cd7c3a79aa8e1ada304fd2639a